### PR TITLE
modal to show detail of img

### PR DESCRIPTION
**Pull request by FedericoChavazza**
## Links:

https://trello.com/c/TToymqGL/14-users-can-see-a-full-screen-picture-of-the-movie

## What & Why:

if you click a card on the movie details, you'll be able to see it fullscreen in a modal .It actually posseses a slider to see images in a more dinamic way too, and a close button to close the modal.

image
image
image

## Risks, Mitigation & Rollback:

- [x] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

File: src/components/common/Button/index.js [hunk 1]
```diff
@@ -2,9 +2,16 @@ import { func, bool, string } from 'prop-types';
 
 import './styles.scss';
 
-const Button = ({ img, children, type = 'button', disabled, handleClick }) => {
+const Button = ({
```
Comment (JulianPasquale) [inline src/components/common/Button/index.js]: Changes seems to be repeated in https://github.com/FedericoChavazza/Movie-App/pull/5.

File: src/components/viewComponents/Slider.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    setImgSelected(imgArray[index + 1]);
+  };
+
+  const goBackward = () => {
+    setImgSelected(imgArray[index - 1]);
+  };
+
+  return (
+    <div>
+      <button disabled={index === 0} onClick={() => goBackward()} className="Slider__left">
```
Comment (JulianPasquale) [inline src/components/viewComponents/Slider.js]:
```suggestion
      <button disabled={index === 0} onClick={goBackward} className="Slider__left">
```

File: src/components/viewComponents/Slider.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <AiFillCaretLeft size={25} />
+      </button>
+      <button
+        className="Slider__right"
+        disabled={index === imgArray.length - 1}
+        onClick={() => goFoward()}
```
Comment (JulianPasquale) [inline src/components/viewComponents/Slider.js]:
```suggestion
        onClick={goFoward}
```

File: src/pages/MovieInfo.js [hunk 4]
```diff
@@ -79,10 +90,16 @@ export function MovieInfo() {
         <div ref={scrollRef} className="MovieInfo__extraImgs">
           {' '}
           {movie?.extraImgs.map((img, i) => {
-            return <img src={img} alt={i} key={i} />;
+            // eslint-disable-next-line jsx-a11y/click-events-have-key-events
```
Comment (JulianPasquale) [inline src/pages/MovieInfo.js]: Can we not disable this rule? What do we need to follow it?

File: src/pages/MovieInfo.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
```
Comment (JulianPasquale) [inline src/pages/MovieInfo.js]: Can we not disable this rule? What do we need to follow it?

Comment (FedericoChavazza) [reply src/pages/MovieInfo.js:1]: if i dont up this thing, this happens, 
image

kind of contradictory

Comment (FedericoChavazza) [reply src/pages/MovieInfo.js]: its asking for a onKeydown for the img, kind of weird, if i put this property, i get this next error 
image
which is kinda weird, the only solution i came up with was this one. Is kind of contradictory but it works somehow
image

File: src/pages/MovieInfo.js [hunk 6]
```diff
@@ -22,6 +27,11 @@ export function MovieInfo() {
 
   const scrollRef = useHorizontalScroll();
 
+  const imageHandler = (e, img) => {
```
Comment (JulianPasquale) [inline src/pages/MovieInfo.js]: Why are you adding a `e` params that will always receive `null`? I think we can only receive `img`

**Review by JulianPasquale — APPROVED** [review]

File: src/components/modalComponent/Modal.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  const modalRef = useRef(null);
+  useEffect(() => {
+    if (show) {
+      modalRef.current.classList.add('visible');
+    } else {
+      modalRef.current.classList.remove('visible');
+    }
+  }, [show, backdropStyle]);
+  return (
+    <>
```
Comment (martoesses) [inline src/components/modalComponent/Modal.js]: In this case is not necessary the fragment because you have a div container

**Review by martoesses — APPROVED** [review]

**Pull request merged**